src/library/scala/collection/mutable/ArrayBuffer.scala

```diff
@@ -323,7 +323,7 @@ object ArrayBuffer extends StrictOptimizedSeqFactory[ArrayBuffer] {
       else if (targetLen < 0)
         throw new Exception(s"Overflow while resizing array of array-backed collection. Requested length: $targetLen; current length: $arrayLen; increase: ${targetLen - arrayLen}")
 
-    if (targetLen > 0 && targetLen <= arrayLen) -1
+    if (targetLen >= 0 && targetLen <= arrayLen) -1
     else {
       checkArrayLengthLimit()
```

Comment:
Is there any reason why is `checkArrayLengthLimit` a local function? To me it would be more readable if its content would be inlined, it would be easier to see what are various possibilities of `targetLen` and `arrayLen` values. If there are performance reasons, perhaps some comment should be in place?

Reply:
maybe rename it `checkTargetLenInBounds`. There are two parameters and `arrayLen` is the one it does **not** check. But I see you mean the if-else if-else is easier to read.

Reply:
I see that I moved the method to local but did not inline it "all the way". Probably I'm just out of practice with throwing exceptions.